1.5/Source/Thirst_Flavour_Pack/HarmonyPatches/StorytellerComp_RefiringUniqueQuest_Patch.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using HarmonyLib;
+using RimWorld;
+using Verse;
+
+namespace Thirst_Flavour_Pack.HarmonyPatches;
+
+[HarmonyPatch(typeof(StorytellerComp_RefiringUniqueQuest))]
+public static class StorytellerComp_RefiringUniqueQuest_Patch
+{
+    public static int IntervalsPassed => Find.TickManager.TicksGame / 1000;
+    public static Lazy<FieldInfo> generateSkipped = new Lazy<FieldInfo>(() => AccessTools.Field(typeof(StorytellerComp_RefiringUniqueQuest), "generateSkipped"));
+
+    [HarmonyPatch(nameof(StorytellerComp_RefiringUniqueQuest.MakeIntervalIncidents))]
+    [HarmonyPrefix]
+    public static bool MakeIntervalIncidents_Patch(StorytellerComp_RefiringUniqueQuest __instance, ref IEnumerable<FiringIncident> __result, IIncidentTarget target)
+    {
+        StorytellerCompProperties_RefiringUniqueQuest Props = __instance.props as StorytellerCompProperties_RefiringUniqueQuest;
+        IncidentDef incident = Props?.incident;
+
+        if (Props == null || incident is not { defName: "GiveQuest_EndGame_ArchonexusVictory" })
+        {
+            return true;
+        }
+
+        incident = Thirst_Flavour_PackDefOf.MSS_GiveQuest_EndGame_ArchospringVictory;
+        Props.incident = incident;
+        __result = [];
+
+        if (incident.TargetAllowed(target))
+        {
+            List<Quest> questsListForReading = Find.QuestManager.QuestsListForReading;
+            Quest quest = Enumerable.FirstOrDefault(questsListForReading, t => t.root == incident.questScriptDef);
+
+            IncidentParms parms = __instance.GenerateParms(incident.category, target);
+
+            if (quest != null)
+            {
+                if (Props.refireEveryDays >= 0.0 && quest.State != QuestState.EndedSuccess && quest.State != QuestState.Ongoing && quest.State != QuestState.NotYetAccepted &&
+                    quest.cleanupTick >= 0 && IntervalsPassed == (int) (quest.cleanupTick + Props.refireEveryDays * 60000.0) / 1000)
```

Comment:
```suggestion
                    quest.cleanupTick >= 0 && IntervalsPassed >= (int) (quest.cleanupTick + Props.refireEveryDays * 60000.0) / 1000)
```

Potential to get stuck I think. If that tick is skipped due to earlier filters intervals passed may be > your "nextFire". I think this should be `>=` instead of `==` but will leave it for you to review just in case I'm wrong and that gets it stuck in some other way.